src/Make/MakeICal.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @author Marcel Bolten <github@marcelbolten.de>
+ * @copyright 2024 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+
+declare(strict_types=1);
+
+namespace Elabftw\Make;
+
+use DateInterval;
+use DateTimeImmutable;
+use DateTimeInterface;
+use DateTimeZone;
+use Elabftw\Enums\ApiEndpoint;
+use Elabftw\Enums\ApiSubModels;
+use Elabftw\Enums\EntityType;
+use Elabftw\Enums\Scope;
+use Elabftw\Interfaces\StringMakerInterface;
+use Elabftw\Models\Config;
+use Elabftw\Models\Items;
+use Elabftw\Models\Scheduler;
+use Elabftw\Models\Users;
+use Elabftw\Models\Todolist;
+use Elabftw\Models\UnfinishedSteps;
+use Elabftw\Models\Notifications\StepDeadline;
+use Elabftw\Models\Steps;
+use Eluceo\iCal\Domain\Entity\Calendar;
+use Eluceo\iCal\Domain\Entity\Event;
+use Eluceo\iCal\Domain\Entity\Todo;
+use Eluceo\iCal\Domain\ValueObject\Alarm;
+use Eluceo\iCal\Domain\ValueObject\Alarm\DisplayAction;
+use Eluceo\iCal\Domain\ValueObject\Alarm\RelativeTrigger;
+use Eluceo\iCal\Domain\ValueObject\DateTime;
+use Eluceo\iCal\Domain\ValueObject\TimeSpan;
+use Eluceo\iCal\Domain\ValueObject\UniqueIdentifier;
+use Eluceo\iCal\Presentation\Factory\CalendarFactory;
+
+use function array_merge;
+use function date_default_timezone_get;
+use function json_decode;
+use function sprintf;
+use function strlen;
+
+/**
+ * Make an iCal file
+ */
+class MakeICal extends AbstractMake implements StringMakerInterface
+{
+    protected string $contentType = 'text/calendar; charset=utf-8';
+
+    public function __construct(protected string $calendarToken)
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Create an iCal file from events
+     */
+    public function getFileContent(): string
+    {
+        $iCalString = $this->getICalString($this->getEntries($this->getCalendarProperties()));
+        $this->contentSize = strlen($iCalString);
+        return $iCalString;
+    }
+
+    /**
+     * Return a nice name for the file
+     */
+    public function getFileName(): string
+    {
+        return 'eLabFTW-calendar.ics';
```

Comment:
I'm partial to full lowercase for filenames. Also, other exported files are lowercase.

Reply:
👍 Let's keep it consistent in the codebase.